runtime/src/main/scala/akka/grpc/internal/AkkaGrpcClient.scala

```diff
@@ -22,6 +24,32 @@ trait AkkaGrpcClient {
   def closed(): Future[Done]
 }
 
+trait AkkaGrpcClientFactory[T <: AkkaGrpcClient] {
+  def apply(settings: GrpcClientSettings)(implicit mat: Materializer, ex: ExecutionContext): T
+}
+
+
+object AkkaGrpcClientFactory {
+  /**
+   * A function to create an AkkaGrpcClient, bundling its own configuration. These objects are convenient to
+   * pass around as implicit values.
+   */
+  trait Configured[T <: AkkaGrpcClient] {
```

Comment:
Added a type for a "configured" factory, i.e. a factory which is partially applied with its settings, materializer and execution context.